rfc/system/5741-binary-idl-with-message-pack.md

```diff
@@ -0,0 +1,490 @@
+# Binary IDL With MessagePack Bytes
+
+**Authors:**
+
+- [@Han-Ru](https://github.com/future-outlier)
+- [@Yee Hing Tong](https://github.com/wild-endeavor)
+- [@Ping-Su](https://github.com/pingsutw)
+- [@Eduardo Apolinario](https://github.com/eapolinario)
+- [@Haytham Abuelfutuh](https://github.com/EngHabu)
+- [@Ketan Umare](https://github.com/kumare3)
+
+## 1 Executive Summary
+### Literal Value
+Literal Value will be `Binary`.
+
+Use `bytes` in `Binary` instead of `Protobuf struct`.
+
+- To Literal
+
+| Before                            | Now                                          |
+|-----------------------------------|----------------------------------------------|
+| Python Val -> JSON String -> Protobuf Struct | Python Val -> (Dict ->) Bytes -> Binary (value: MessagePack Bytes, tag: msgpack) IDL Object |
+
+- To Python Value
+
+| Before                            | Now                                          |
+|-----------------------------------|----------------------------------------------|
+| Protobuf Struct -> JSON String -> Python Val | Binary (value: MessagePack Bytes, tag: msgpack) IDL Object -> Bytes -> (Dict ->) -> Python Val |
+
+Note: if a python value can't directly be converted to `MessagePack Bytes`, we can convert it to `Dict`, and then convert it to `MessagePack Bytes`.
```

Comment:
Add a note here to say very clearly that there is no JSON in the new type at all.  JSON plays zero part of the new spec.

Reply:
No problem